src/main/java/uk/ac/ic/wlgitbridge/bridge/db/postgres/PostgresDBStore.java

```diff
@@ -149,58 +159,64 @@ public void deleteFilesForProject(String project, String... files) {
       }
     }
     queryBuilder.append(");\n");
-    try (
-      Connection connection = pool.getConnection();
-      PreparedStatement statement = connection.prepareStatement(queryBuilder.toString())) {
-      statement.setString(1, project);
-      for (int i = 0; i < files.length; i++) {
-        statement.setString(i + 2, files[i]);
+    try {
+      Connection connection = connectionFromContext(project);
+      try (PreparedStatement statement = connection.prepareStatement(queryBuilder.toString())) {
+        statement.setString(1, project);
+        for (int i = 0; i < files.length; i++) {
+          statement.setString(i + 2, files[i]);
+        }
+        statement.execute();
       }
-      statement.execute();
-    } catch (Exception e) {
-      throw new RuntimeException("Postgres query error", e);
+    } catch (Throwable t) {
+      throw new RuntimeException("Postgres query error", t);
     }
   };
 
   @Override
   public String getPathForURLInProject(String projectName, String url) {
-    try (
-      Connection connection = pool.getConnection();
-      PreparedStatement statement = connection.prepareStatement(
-        "SELECT path "
-          + "FROM url_index_store "
-          + "WHERE project_name = ? "
-          + "AND url = ?;\n")) {
-      statement.setString(1, projectName);
-      statement.setString(2, url);
-      try (ResultSet rs = statement.executeQuery()) {
-        if (rs.next()) {
-          String path = rs.getString(1);
-          return path;
-        } else {
-          return null;
+    try {
+      Connection connection = connectionFromContext(projectName);
+      try (PreparedStatement statement = connection.prepareStatement(
+              "SELECT path "
+                      + "FROM url_index_store "
+                      + "WHERE project_name = ? "
+                      + "AND url = ?;\n")) {
+        statement.setString(1, projectName);
+        statement.setString(2, url);
+        try (ResultSet rs = statement.executeQuery()) {
+          if (rs.next()) {
+            String path = rs.getString(1);
+            return path;
+          } else {
+            return null;
+          }
         }
       }
-    } catch (Exception e) {
-      throw new RuntimeException("Postgres query error", e);
+    } catch (Throwable t) {
+      throw new RuntimeException("Postgres query error", t);
     }
   };
 
   @Override
   public String getOldestUnswappedProject() {
-    try (
-      Connection connection = pool.getConnection();
-      Statement statement = connection.createStatement()) {
-      try (ResultSet rs = statement.executeQuery(
-          "SELECT name FROM projects" +
-            " ORDER BY last_accessed ASC" +
-            " LIMIT 1;\n")) {
-        if (rs.next()) {
-          String project = rs.getString(1);
-          return project;
-        } else {
-          return null;
+    try {
+      Connection connection = connectionFromPool();
+      connection.setAutoCommit(true);
```

Comment:
Related to my questions above. Here, we set autocommit before executing the statement and the connection is closed explicitly at the end.